rest_framework_swagger/docgenerator.py

```diff
@@ -338,10 +338,10 @@ def _get_serializer_fields(self, serializer):
                     has_many = isinstance(field, ListSerializer)
                 if has_many:
                     f['type'] = 'array'
-                    if data_type in BaseMethodIntrospector.PRIMITIVES:
-                        f['items'] = {'type': data_type}
-                    else:
+                    if field_serializer:
                         f['items'] = {'$ref': field_serializer}
+                    elif data_type in BaseMethodIntrospector.PRIMITIVES:
+                        f['items'] = {'type': data_type}
```

Comment:
Are there any cases where field_serializer is None and data_type not in BaseMethodIntrospector.PRIMITIVES?